packages/extensions/openapi-to-typespec/src/resource/utils.ts

```diff
@@ -79,8 +79,21 @@ export function isSingleton(set: OperationSet): boolean {
 export function pathIncludes(path1: string, path2: string): boolean {
   const lowerPath1 = path1.toLowerCase();
   const lowerPath2 = path2.toLowerCase();
-  // TO-DO: escape the variable case
-  return lowerPath1.includes(lowerPath2);
+  const segments1 = lowerPath1.split("/");
+  const segments2 = lowerPath2.split("/");
+  if (segments2.length > segments1.length) return false;
+
+  for (let index = 0; index < segments2.length; ++index) {
+    if (isVariable(segments1[index])) {
+      if (!isVariable(segments2[index])) return false;
```

Comment:
Is it because the variable name can be different from the paths?

Reply:
Yes